retention/options.go

```diff
@@ -64,12 +71,42 @@ func NewOptions() Options {
 		blockSize:                        defaultBlockSize,
 		bufferFuture:                     defaultBufferFuture,
 		bufferPast:                       defaultBufferPast,
-		bufferDrain:                      defaultBufferDrain,
 		dataExpiry:                       defaultDataExpiry,
 		dataExpiryAfterNotAccessedPeriod: defaultDataExpiryAfterNotAccessedPeriod,
 	}
 }
 
+func (o *options) Validate() error {
+	if o.bufferFuture < 0 {
+		return errBufferFutureNonNegative
+	}
+	if o.bufferPast < 0 {
+		return errBufferPastNonNegative
+	}
+	if o.blockSize <= 0 {
+		return errBlockSizePositive
+	}
+	if o.bufferFuture >= o.blockSize {
+		return errBufferFutureTooLarge
+	}
+	if o.bufferPast >= o.blockSize {
+		return errBufferPastTooLarge
+	}
+	if o.retentionPeriod < o.blockSize {
+		return errRetentionPeriodTooSmall
+	}
+	return nil
+}
+
+func (o *options) Equal(value Options) bool {
```

Comment:
Following convention of `Equal` from `ts.ID.Equal()`. Feels weird to have different names for the same semantic thing (comparing equality) based on the interface representing a single or multiple items.